editoast/src/views/layers/mod.rs

```diff
@@ -131,7 +132,7 @@ async fn layer_view(
     Ok(Json(ViewMetadata {
         data_type: "vector".to_owned(),
         name: layer_slug.to_owned(),
-        promote_id: json!({layer_slug: layer.id_field}),
+        promote_id: HashMap::from([(layer_slug, layer.id_field.clone().unwrap_or_default())]),
```

Comment:
```suggestion
        promote_id: HashMap::from([(layer_slug, layer.id_field.unwrap_or_default())]),
```

If that conflicts with what's below you should unpack the struct beforehand.